### DISCUSSION TITLE
How can I add programmatically delays in a connection?

### DISCUSSION THREAD

**Discussion by apaikan** [robotology/community#99] [Q&A] [answered]
By @gregoire-pointeau 

Hello.
Here is my problem.
I have a module (let's call it A) that streams information to a second module (let's call it B).
I would like to introduce some delay between the A and B.
My idea would be to create a module C in the middle, that would get the Bottle from A, wait, and forward it to B.
I wonder if something like that already exists in port monitor or something equivalent ?

Thank you in advance !

**Comment by apaikan** [comment, marked as answer]
Hi @gregoire-pointeau

Yes! `portmonitor` is the correct way to go and here is an example showing exactly what you need: https://github.com/robotology/yarp/blob/master/example/portmonitor/port_rate/rate.lua.

Here is also the documentation regarding `portmonitor`, if you have not been using it yet: http://www.yarp.it/portmonitor.html.

**Comment by gregoire-pointeau** [comment]
Hi @apaikan , thank you for your answer.
I'll have a look at it, and I think come back to you soon!

**Comment by pattacini** [comment]
For the time being we can close this.
@gregoire-pointeau feel free to open it again when deemed necessary.
